### PR TITLE
Handle special case of xssProtection

### PR DESCRIPTION
**Pull request by SiBorea**
## What's changed?
When it comes to `xssProtection`, the recipe will generate code like `protection ->protection.xssProtectionEnabled(true)`. However, `xssProtectionEnabled()` is removed in spring-security-config-6+. xssProtection is set enabled as default and disabled on demand.

### Checklist
- [✅] I've added unit tests to cover both positive and negative cases
- [✅] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [✅] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java [hunk 1]
```diff
@@ -38,79 +38,157 @@ void simpleContentSecurityPolicy() {
...
+                              .headers()
+                              .xssProtection(protection -> protection.disable());
+                  }
+              }
+              """
+          )
+        );
     }
-}
+
```
Comment (timtebeek) [inline src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java:154]: There's potentially an edge case related to the`XssProtectionSpec.headerValue` added in Spring Security 5.8. Not a lot of folks are likely to use that, but figured call it out here as something that we either want to be explicit about not handling yet, or handle and test as well.

File: src/main/java/org/openrewrite/java/spring/boot2/ConvertToSecurityDslVisitor.java [hunk 2]
```diff
@@ -127,6 +128,15 @@ private static String generateParamNameFromMethodName(String n) {
     private J.Lambda createLambdaParam(String paramName, JavaType paramType, List<J.MethodInvocation> chain) {
         J.Identifier param = createIdentifier(paramName, paramType);
         J.MethodInvocation body = unfoldMethodInvocationChain(createIdentifier(paramName, paramType), chain);
+        // Special case for xssProtectionEnabled method
+        if (XSS_PROTECTION_ENABLED.matches(body)) {
+            if (J.Literal.isLiteralValue(body.getArguments().get(0), false)) {
+                body = body.withName(body.getName().withSimpleName("disable")).withArguments(null);
+            } else {
+                // Enabled by default; but returning `null` will cause issues, so we use `and()` as a placeholder
+                body = body.withName(body.getName().withSimpleName("and")).withArguments(null);
```
[the remark this answers is not in the document]
Comment (SiBorea) [reply src/main/java/org/openrewrite/java/spring/boot2/ConvertToSecurityDslVisitor.java:137]: It's here: https://github.com/spring-projects/spring-security/blob/ca1f891f256738ffa765cdd5df02070127070e3e/config/src/main/java/org/springframework/security/config/annotation/web/configurers/HeadersConfigurer.java#L881

File: src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java (hunk 1 above)
Comment (SiBorea) [reply src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java:162]: It's supported by the origin recipe. I added a new test for it

File: src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java [hunk 3]
```diff
@@ -38,37 +46,154 @@ void simpleContentSecurityPolicy() {
...
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+              import org.springframework.security.web.header.writers.XXssProtectionHeaderWriter;
+
+              @Configuration
+              public class WebSecurityConfig {
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                          .headers()
+                          .xssProtection()
+                          .headerValue(XXssProtectionHeaderWriter.HeaderValue.DISABLED);
```
Comment (timtebeek) [inline src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java:179]: When I add this line here
```suggestion
                          .xssProtection()
                          .xssProtectionEnabled(true)
                          .headerValue(XXssProtectionHeaderWriter.HeaderValue.DISABLED);
```
Then replacement ends up containing
```
diff --git a/WebSecurityConfig.java b/WebSecurityConfig.java
index 08c2b14..16e345b 100644
--- a/WebSecurityConfig.java
+++ b/WebSecurityConfig.java
@@ -7,7 +7,8 @@ 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .headers()
-                .xssProtection(protection -> protection
+                .xssProtection(protection -> protection
+                        .xssProtectionEnabled(true)
                         .headerValue(XXssProtectionHeaderWriter.HeaderValue.DISABLED));
     }
 }
```
Whereas that `xssProtectionEnabled(true)` should be removed right? Or replaced with `.and()` ?

Comment (SiBorea) [reply src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java]: Yes. Forgot to test the invocation chain. Made a little fix

**Review by timtebeek — APPROVED** [review]
Thanks a lot for the improved handling here @SiBorea , and your continued involvement throughout. Much appreciated!

**Pull request merged**